DsgTools/core/DSGToolsProcessingAlgs/Algs/OtherAlgs/exportFeaturesByAttributeAlgorithm.py

```diff
@@ -0,0 +1,195 @@
+
+    def initAlgorithm(self, config=None):
+        self.addParameter(
+            QgsProcessingParameterVectorLayer(
+                self.INPUT,
+                self.tr('Input layer'),
+                [QgsProcessing.TypeVector]
+            )
+        )
+
+        self.addParameter(
+            QgsProcessingParameterField(
+                self.FIELD,
+                self.tr('Selection attribute'),
+                parentLayerParameterName=self.INPUT
+            )
+        )
+
+        self.addParameter(
+            QgsProcessingParameterEnum(
+                self.OPERATOR,
+                self.tr('Operator'),
+                options=self.OPERATORS,
+                defaultValue=0
+            )
+        )
+
+        self.addParameter(
+            QgsProcessingParameterString(
+                self.VALUE,
+                self.tr('Value'),
+                optional=True
+            )
+        )
+
+        self.addParameter(
+            QgsProcessingParameterFeatureSink(
+                self.OUTPUT,
+                self.tr('Output layer')
+            )
+        )
+
+    def processAlgorithm(self, parameters, context, feedback):
+        layer = self.parameterAsVectorLayer(parameters, self.INPUT, context)
+        fieldName = self.parameterAsString(parameters, self.FIELD, context)
+        operator = self.OPERATORS[self.parameterAsEnum(parameters, self.OPERATOR, context)]
+        value = self.parameterAsString(parameters, self.VALUE, context)
+
+        if layer is None:
+            feedback.pushInfo("Camada inválida ou não encontrada.")
+            return {self.OUTPUT: None}
+
+        if fieldName not in layer.fields().names():
+            feedback.pushInfo(f"O atributo '{fieldName}' não foi encontrado na camada. Nenhuma exportação realizada.")
+            return {self.OUTPUT: None}
+
+        # Construindo a expressão de seleção
+        field_ref = QgsExpression.quotedColumnRef(fieldName)
+        if operator == 'is null':
+            expression_string = f"{field_ref} IS NULL"
+        elif operator == 'is not null':
+            expression_string = f"{field_ref} IS NOT NULL"
+        elif operator == 'begins with':
+            expression_string = f"{field_ref} LIKE '{value}%'"
+        elif operator == 'contains':
+            expression_string = f"{field_ref} LIKE '%{value}%'"
+        elif operator == 'does not contain':
+            expression_string = f"{field_ref} NOT LIKE '%{value}%'"
+        else:
+            quoted_val = QgsExpression.quotedValue(value)
+            expression_string = f"{field_ref} {operator} {quoted_val}"
+
+        expression = QgsExpression(expression_string)
+        if expression.hasParserError():
+            feedback.pushInfo(f"Erro na expressão: {expression.parserErrorString()}")
+            return {self.OUTPUT: None}
+
+        # Configurando o sink para a camada de saída
+        (sink, dest_id) = self.parameterAsSink(
```

Comment:
Essa linha tem que ir lá pro início, para usar o dest_id nos retornos. Retorno None dá erro em model, não é o caso